xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1544,6 +1547,17 @@ static CdsUpdate parseCluster(Cluster cluster, Set<String> retainedEdsResources,
       updateBuilder.ringHashLbPolicy(minRingSize, maxRingSize);
     } else if (cluster.getLbPolicy() == LbPolicy.ROUND_ROBIN) {
       updateBuilder.roundRobinLbPolicy();
+    } else if (cluster.getLbPolicy() == LbPolicy.LEAST_REQUEST) {
```

Comment:
The equivalent condition for ring_hash was there even when it was disabled by default.
We intend to try using this LB during experimental status and would like some way to enable it programatically that doesn't involve the environment variable (since we can't set envs in java).
For the provider I was thinking of doing something like this to make it available in our service framework.

```java
import io.grpc.xds.LeastRequestLoadBalancerProvider

// Temporary least_request_experimental provider that makes it enabled by default.
public final class TemporaryLeastRequestLoadBalancerProvider extends LoadBalancerProvider {

  private final LoadBalancerProvider grpcProvider;
  
  public TemporaryLeastRequestLoadBalancerProvider() {
    this.grpcProvider = new LeastRequestLoadBalancerProvider();
  }
  
  @Override
  public LoadBalancer newLoadBalancer(LoadBalancer.Helper helper) {
    return grpcProvider.newLoadBalancer(helper);
  }
  
  @Override
  public boolean isAvailable() {
    return true;
  }
  ...
}

```

But an approach like that wouldn't work if this conditional is disabled.
If you still want this conditional disabled by default for now perhaps also using a Java property would help us achieve what we want.
Maybe something like:

```java
private static final boolean enableLeastRequest =
      Boolean.parseBoolean(System.getenv("GRPC_EXPERIMENTAL_ENABLE_LEAST_REQUEST")) ||
      Boolean.parseBoolean(System.getProperty("io.grpc.xds.gRPCExperimentalEnableLeastRequest")
```

wdyt?

Reply:
I thought more about this and decided that a system property would be easiest here.
Then we don't need any `TemporaryLeastRequestLoadBalancerProvider` that relies on an `@Internal` class and we can disable this condition completely when 
GRPC_EXPERIMENTAL_ENABLE_LEAST_REQUEST=false. 
I did a bit differently from the example above to make sure that GRPC_EXPERIMENTAL_ENABLE_LEAST_REQUEST is used if set and io.grpc.xds.experimentalEnableLeastRequest otherwise if that is set.
This means it should be consistent with the gRFC and other implementations.